src/main/java/cz/cvut/kbss/termit/service/business/SparqlExternalVocabularyService.java

```diff
@@ -0,0 +1,223 @@
+            if (line.hasBinding("nazev_slovniku_cs")) {
+                labels.put("cs", line.getBinding("nazev_slovniku_cs").getValue().stringValue());
+            } else {
+                labels.put("cs", uri.toString());
+            }
+            // add en label if available
+            if (line.hasBinding("nazev_slovniku_en")) {
+                labels.put("en", line.getBinding("nazev_slovniku_en").getValue().stringValue());
+            } else {
+                labels.put("en", uri.toString());
+            }
+            MultilingualString label = new MultilingualString(labels);
+            response.add(new RdfsResource(uri, label, new MultilingualString(), ""));
+        }
+        return response;
+    }
+
+    /**
+     * Imports multiple vocabularies from external source.
+     *
+     * @param vocabularyIris List of iris of vocabularies that shall be imported.
+     * @return first imported Vocabulary
+     */
+    @Transactional
+    @PreAuthorize("@vocabularyAuthorizationService.canCreate()")
+    @Override
+    public Vocabulary importFromExternalUris(List<String> vocabularyIris) {
+        Vocabulary firstImportedVocabulary = null;
+
+        for (String vocabularyIri : vocabularyIris) {
+            try {
+                LOG.trace("Starting import of external vocabulary {}", vocabularyIri);
+                InputStream newVocabulary = downloadExternalVocabulary(vocabularyIri);
+                if (newVocabulary != null) {
+                    URI uri = URI.create(vocabularyIri);
+                    Vocabulary vocabulary = repositoryService.importVocabulary(uri, RDFFormat.TURTLE.getDefaultMIMEType(), newVocabulary);
+
+                    // add types
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_pouze_pro_cteni);
+                    vocabulary.addType(cz.cvut.kbss.termit.util.Vocabulary.s_c_externi);
+
+                    final AccessControlList acl = aclService.createFor(vocabulary);
+                    vocabulary.setAcl(acl.getUri());
+
+                    if (repositoryService.find(uri).isEmpty()) { // The vocabulary is new
+                        eventPublisher.publishEvent(new VocabularyCreatedEvent(this, vocabulary.getUri()));
+                    }
+
+                    LOG.trace("Vocabulary {} import was successful.", vocabularyIri);
+
+                    if (firstImportedVocabulary == null) {
+                        firstImportedVocabulary = vocabulary;
+                    }
+                }
+            } catch (QueryEvaluationException | RepositoryException ex) {
+                LOG.error(ex.getMessage());
+            }
+        }
+        
+        return firstImportedVocabulary;
+    }
+
+    private InputStream downloadExternalVocabulary(String vocabularyIri) throws QueryEvaluationException, RepositoryException{
+        
+        SPARQLRepository sparqlRepo = initSparqlRepository();
+        
+        RepositoryConnection conn = sparqlRepo.getConnection();
```

Comment:
Missing error handling/proper shutdown here. Since exception is caught in the caller, the connection and repository will be left open.